eng/common/TestResources/New-TestResources.ps1

```diff
@@ -379,6 +388,10 @@ foreach ($templateFile in $templateFiles) {
         "$($serviceDirectoryPrefix)STORAGE_ENDPOINT_SUFFIX" = $context.Environment.StorageEndpointSuffix;
     }
 
+    foreach ($ev in $EnvironmentVariables) {
+        $deploymentOutputs[$ev.Name] = $ev.Value
```

Comment:
I worry a bit this could overwrite variables atop. Are we reasonably certain that wouldn't happen? Otherwise, should we maybe call `$deploymentOutputs.Add($ev.Name, $ev.Value)` here to make sure it throws (alerting us of a problem) instead?

Reply:
...and worried only about these variables given they are both calculated outputs (well, some are inputs, but credential-related ones). Overwriting those could be disastrous, but if that's unlikely this is perfectly fine.

Reply:
It just depends which we think has higher priority. I was thinking we'd want to overwrite any values, as if they are the subscription config I'd assume they are known overrides necessary for a cloud to work. I'm fine with either logging or throwing though just to make sure this behavior isn't hidden.